qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowserplotinteraction.py

```diff
@@ -0,0 +1,308 @@
+            self.update_current_handler_guess()
+        if self.guess_all_line:
+            self.update_guess_all()
+
+    @Slot()
+    def slot_for_function_removed(self):
+        """
+        Responds to a function being removed from the FitPropertyBrowser,
+        the function is found from the currently selected handler
+        If the function, or guess all is plotted the plot is updated accordingly
+        """
+        if not self.fit_browser.currentHandler():
+            return
+        prefix = self.fit_browser.currentHandler().functionPrefix()
+        self.remove_current_guess()
+        self._shift_stored_prefixes(prefix)
+        if self.guess_all_line:
+            self.update_guess_all()
+
+    def plot_current_guess(self):
+        """
+        Plot the guess workspace of the currently selected function
+        """
+        fun = self.fit_browser.currentHandler().ifun()
+        ws_name = self.fit_browser.workspaceName()
+        if fun == '' or ws_name == '':
+            return
+        out_ws_name = self._get_current_prefixed_function_name()
+
+        line = self._plot_guess_workspace(ws_name, fun, out_ws_name)
+
+        if line:
+            self.guess_lines[self._get_current_prefixed_function_name()] = line
+
+    def evaluate_function(self, ws_name, fun, out_ws_name):
+        """
+        Evaluates the guess workspace for the input workspace and function
+        :param ws_name: Name of the workspace in the ADS
+        :param fun: Function to be evaluated
+        :param out_ws_name: Output workspace name
+        :return: Output guess workspace
+        """
+        ws_index = self.fit_browser.workspaceIndex()
+        startX = self.fit_browser.startX()
+        endX = self.fit_browser.endX()
+        workspace = AnalysisDataService.retrieve(ws_name)
+        alg = AlgorithmManager.createUnmanaged('EvaluateFunction')
+        alg.setChild(True)
+        alg.initialize()
+        alg.setProperty('Function', fun)
+        alg.setProperty('InputWorkspace', ws_name)
+        if isinstance(workspace, ITableWorkspace):
+            alg.setProperty('XColumn', self.fit_browser.getXColumnName())
+            alg.setProperty('YColumn', self.fit_browser.getYColumnName())
+            if self.fit_browser.getErrColumnName():
+                alg.setProperty('ErrColumn', self.fit_browser.getErrColumnName())
+        else:
+            alg.setProperty('WorkspaceIndex', ws_index)
+        alg.setProperty('StartX', startX)
+        alg.setProperty('EndX', endX)
+        alg.setProperty('OutputWorkspace', out_ws_name)
+        alg.execute()
+
+        return alg.getProperty('OutputWorkspace').value
+
+    def plot_guess_all(self):
+        """
+        Plot the guess workspace for the full function
+        """
+        fun = self.fit_browser.getFittingFunction()
+        ws_name = self.fit_browser.workspaceName()
+        if fun == '' or ws_name == '':
+            return
+        out_ws_name = '{}_guess'.format(ws_name)
```

Comment:
Could we use [f-strings](https://realpython.com/python-f-strings/) throughout here for better readability?

Reply:
Yep, good point 👍